portals/admin/source/src/app/components/AdvancedSettings/TenantConfSave.jsx

```diff
@@ -0,0 +1,240 @@
+    },
+    helperText: {
+        color: green[600],
+        fontSize: theme.spacing(1.6),
+        marginLeft: theme.spacing(1),
+    },
+    infoBox: {
+        marginTop: theme.spacing(1),
+        marginBottom: theme.spacing(2),
+    },
+    buttonBox: {
+        marginTop: theme.spacing(2),
+        marginBottom: theme.spacing(2),
+        marginLeft: theme.spacing(1),
+    },
+    saveButton: {
+        marginRight: theme.spacing(2),
+    },
+}));
+
+/**
+ * Reducer
+ * @param {JSON} state The second number.
+ * @returns {Promise}
+ */
+function reducer(state, { field, value }) {
+    switch (field) {
+        case 'tenantConf':
+            return { ...state, [field]: value };
+        case 'tenantConfSchema':
+            return { ...state, [field]: value };
+        case 'editDetails':
+            return value;
+        default:
+            return state;
+    }
+}
+
+/**
+ * Render a list
+ * @returns {JSX} Header AppBar components.
+ */
+function TenantConfSave(props) {
+    const classes = useStyles();
+    const {
+        updateList, history,
+    } = props;
+    const intl = useIntl();
+    const [initialState, setInitialState] = useState({
+        tenantConf: '',
+        tenantConfSchema: '',
+    });
+    const [state, dispatch] = useReducer(reducer, initialState);
+    const {
+        tenantConf, tenantConfSchema,
+    } = state;
+    const restApi = new API();
+
+    useEffect(() => {
+        let tenantConfVal;
+        let tenantConfSchemaVal;
+        restApi.tenantConfSchemaGet().then((result) => {
+            tenantConfSchemaVal = result.body;
+            dispatch({ field: 'tenantConfSchema', value: tenantConfSchemaVal });
+        });
+        restApi.tenantConfGet().then((result) => {
+            tenantConfVal = JSON.stringify(result.body, null, '\t');
+            dispatch({ field: 'tenantConf', value: tenantConfVal });
+        });
+        setInitialState({
+            tenantConf: tenantConfVal,
+            tenantConfSchema: tenantConfSchemaVal,
+        });
+    }, []);
+
+    const editorWillMount = (monaco) => {
+        const schemaVal = state.tenantConfSchema;
+        monaco.languages.json.jsonDefaults.setDiagnosticsOptions({
+            completion: true,
+            validate: true,
+            format: true,
+            schemas: [{
+                uri: 'http://myserver/foo-schema.json',
+                fileMatch: ['*'],
+                schema: schemaVal,
+            }],
+        });
+    };
+
+    const tenantConfOnChange = (newValue) => {
+        dispatch({ field: 'tenantConf', value: newValue });
+    };
+
+    const cancelUpdate = () => {
+        let tenantConfVal;
+        restApi.tenantConfGet().then((result) => {
+            tenantConfVal = JSON.stringify(result.body, null, '\t');
+            const editState = {
+                tenantConf: tenantConfVal,
+            };
+            dispatch({ field: 'editDetails', value: editState });
+        });
+        setInitialState({
```

Comment:
Same as above have no effect on the reducer's initial state, Have to use the dispatcher instead 

is this `cancelUpdate` work as expected? is it reset the reducer's state to initial value ?

Reply:
yes, currently it reset to initial value without issues

Reply:
will change accordingly

Reply:
resolved